.github/copilot-instructions.md

```diff
@@ -0,0 +1,139 @@
+# GitHub Copilot Instructions for autorest.typescript
+
+This repository focuses on the TypeSpec TypeScript generator, which generates TypeScript client libraries from [TypeSpec](https://typespec.io/) specifications. TypeSpec is the recommended approach for describing cloud service APIs and generating client libraries.
+
+> **Note:** While this repository also contains the AutoRest TypeScript generator for OpenAPI/Swagger specifications, we recommend using TypeSpec for new projects. The AutoRest generator is in maintenance mode and will be deprecated in the future.
+
+## Project Structure
+
+- `packages/typespec-ts/` - TypeSpec TypeScript generator
+
+  - Generates TypeScript clients from TypeSpec definitions
+  - Supports both RLC and Modular SDK generation
+  - RLC provides low-level REST client operations
+  - Modular SDK offers modern, enhanced developer experience
+  - Uses TypeSpec compiler for processing
+
+- `packages/rlc-common/` - Common utilities for REST Level Client generation
+
+  - Core RLC patterns and utilities
+  - Shared code for TypeSpec RLC implementation
+  - Location for RLC changes and improvements
+
+- `packages/typespec-test/` - Test utilities and fixtures for TypeSpec code generation
+
+  - Test framework for TypeSpec generator
+  - Sample TypeSpec definitions
+  - Validation utilities
+  - Test fixtures and helpers
+
+- `packages/autorest.typescript/` - AutoRest TypeScript generator
+  > **Note:** In maintenance mode, will be deprecated
+
+## AutoRest TypeScript Generator
+
+The AutoRest TypeScript generator (`autorest.typescript`) transforms OpenAPI/Swagger specifications into TypeScript client libraries. It supports two client styles:
+
+1. **REST Level Client (RLC)** - Low-level mapping to REST operations
+2. **High Level Client (HLC)** - Higher-level abstraction layer
+
+For detailed documentation on the AutoRest generator, please refer to the [AutoRest documentation](https://github.com/Azure/autorest).
+
+## TypeSpec Test Framework
+
+The TypeSpec test framework (`typespec-test`)provides utilities and test fixtures for validating TypeSpec code generation. Here are the key aspects to consider:
+
+## TypeSpec TypeScript Generator
+
+The TypeSpec TypeScript generator (`typespec-ts`) generates TypeScript client libraries from [TypeSpec](https://typespec.io/) specifications. TypeSpec is a language specifically designed for API development that:
+
+- Provides a more intuitive way to describe APIs
+- Supports rich type systems and inheritance
+- Enables better code organization through namespaces and imports
+- Allows custom decorators for metadata and behavior
+- Generates consistent API artifacts (clients, documentation, etc.)
+
+The generator supports two distinct SDK styles:
+
+### SDK Generation Styles
+
+1. **REST Level Client (RLC)**
```

Comment:
We talk about RCL and Modular at te beginning. I wonder if it would be better if we added a Key concepts section at the beginning where we describe these. Then here we just say it can generate: - Rlc libraries - Modular Libraries